Doc/c-api/hash.rst

```diff
@@ -41,6 +45,27 @@ See also the :c:member:`PyTypeObject.tp_hash` member.
    .. versionadded:: 3.4
 
 
+Functions
+^^^^^^^^^
+
+.. c:function:: int Py_HashDouble(double value, Py_hash_t *result)
+
+   Hash a C double number.
+
+   * Set *\*result* to the hash value and return ``1`` on success.
+   * Set *\*result* to ``0`` and return ``0`` if the hash value cannot be
+     calculated. For example, if *value* is not-a-number (NaN).
+
+   *result* must not be ``NULL``.
+
+   .. note::
+      Only rely on the function return value to distinguish the "not-a-number"
+      case. *\*result* can be ``0`` if *value* is finite. For example,
+      ``Py_HashDouble(0.0, &result)`` sets *\*result* to 0.
+
```

Comment:
Maybe we can add a recipe suggesting to use `Py_HashPointer()` if the function returns 0?

```c
Py_hash_t hash_(double value, PyObject *obj)
{
    Py_hash_t hash;
    if (Py_HashDouble(value, &hash) == 0) {
        hash = Py_HashPointer(obj);
    }
    return hash;
}
```

Reply:
What's the use case for that?

Reply:
If you want to exactly match the way a Python `double` is hashed, you need to construct the `PyDouble` object and then you can call `PyObject_Hash` on it. If you don't have a `PyDouble` object, then what should you pass to `hash_`?

Reply:
Hum, it's mostly to replace existing code using _Py_HashDouble(). I updated What's New in Python 3.13 to suggest a recipe to replace it.